### PR TITLE
News week homepage (responsive design)

### PR DESCRIPTION
**Pull request by Bishoy-Samwel**
We are tasked to clone NewsWeek homepage using Bootstrap classes.

Please leave a comment where modifications are necessary.

### CODE REVIEW

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Invalid 🛑 

Hello, you've made great progress, but I can't review your work unless the requirements below are met 👇 

- [x] The project should have a descriptive README.md ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
* [x] Pull Request has a meaningful title and a brief description ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_prs.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))



Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Invalid 🛑 

Hello, you've made great progress, but I can't review your work unless the requirements below are met 👇 

- [x] The project should have a descriptive README.md, yours is not descriptive enough. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))  - feel free to use [this template](https://github.com/microverseinc/readme-template)


Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Review by PhillipUg — CHANGES_REQUESTED** [review]
Hey team, 👋

You have done an amazing job implementing the requirements for this project.

However, there are still some changes 👇 you need to implement before moving on to your next milestone 🔍

### Changes Required ♻️ 

- [x] First of all, fix all linter errors. Kindly ensure to fix all linter errors before submitting for review in all your future projects

#### Header
- [x] The Newsweek logo has a link
    -  In desktop it is centered
    -  In tablet and mobile, it is aligned to the left
* [x] The Navbar contains a date only in desktop
* [x] Login and Subscribe have links
  - In mobile, the Login button changes into an icon
* [x] The main navigation (menu)
  - Uses a nav tag
  - Each option has a link
  - It has a form with the search field
  - In mobile and table the menu hides, and show a hamburger button
* [x] The size of the elements change in desktop, tablet and mobile

#### Footer

* [x] The Newsweek logo has a link
* [x] The social media icons have links
* [x] The content disposition changes to the correct place in desktop, tablet and mobile

#### Please note that this request will be approved in the dashboard and you don’t need to submit it for re-review or merge this request into the main branch please make the suggested changes and move on to the next milestone. When you're done with the next milestone, submit this request again, don't create a new pull request. You are to keep submitting this same pull request until project completion.

If you have any questions, feel free to send me a message through Slack as **Phillip Musiime** or
 <a href="mailto:phillipmusiime@gmail.com"> gmail</a> &nbsp; &nbsp; [Twitter Follow](https://twitter.com/intent/follow?screen_name=Phillip_Ug) &nbsp; &nbsp; [GitHub followers](https://github.com/PhillipUg)

**Comment by euqueme** [comment]
Hello guys, :wave:  I see that the previous reviewer marked Milestone 1 as requested changes instead of Approved on the Dashboard, I see that you already worked on the whole project so that's really awesome keep it up! :muscle:  but since I cannot perform a re-review for this milestone I'm gonna approve it so you can submit for a code review for Milestone 2

I'm sorry for the bothers

Happy coding! :rocket:

**Comment by uryelah** [comment]
### Invalid submission 🛑 

Hey Guys, I'm marking the submission for Milestone 3 as **INVALID** since you have not yet received the review for Milestone 2.

Please **wait until you request for and get Milestone 2 reviewed** and then submit the review for Milestone 3 again 😃 

---

Happy coding 🍀

File: index.html [hunk 1]
```diff
@@ -0,0 +1,2263 @@
...
+        </nav>
+      </nav>
+      <!-- first section 
+            [FEATURED STORIES], 
+            [TOP STORY, MY TURN, CULTURE & TRAVEL, MORE STORIES] 
+            [THE DEBATE, OPINION]-->
+      <div class="container">
+        <div class="row">
+          <!-- col-1 FEATURED STORIES -->
+          <div class="col-lg-4 col-md-5 col-sm-12">
```
Comment (jubaan) [inline index.html]: - [ ] Use semantic HTML for `<main>`, `<article>`, `<section>`, `<h1>` to `<h6>` for titles, minimize the use of `div` containers they are generic and make it difficult to navigate through the HTML file. Semantic tags improve readability and reduces the complexity of the code. Review your HTML and apply semantic HTML

File: style.css [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+#img-2 {
+  bottom: 0;
+  right: 0;
+}
+
+#img-1 {
+  left: 0;
+  top: 0;
+}
+
```
Comment (jubaan) [inline style.css]: - [x] Remove ID, use custom CSS selectors to point out to the elements. Review all style.css file it should not contain ID, while it works on solving the issues it creates a difficult code to maintain in the long run.

File: style.css [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  #login-txt {
+    display: none;
+  }
+}
+
+@media (max-width: 450px) {
+  #footer-column {
+    display: flex;
+    flex-direction: column;
+    justify-content: left;
```
Comment (jubaan) [inline style.css]: - Use bootstrap classes for **columns, flex, flex-direction, justify-content, align-content, display, position, font-weight, text-align**... etc. Review the style.css file and remove all CSS rules that have a Bootstrap class and is useful for the project.

File: style.css [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  #navbar-md-sm {
+    display: none;
+  }
+}
+
+@media (max-width: 991px) {
+  #first-header,
+  #second-header,
+  #third-header {
+    display: none !important;
```
Comment (jubaan) [inline style.css]: - [x] Remove all `!important` from the CSS stylesheet. don't override classes, and find a way whether with Bootstrap classes or with custom CSS is Bootstrap doesn't have a class for it.

**Review by jubaan — CHANGES_REQUESTED** [review]
# Status: Approved with OBSERVATIONS :recycle: 

There are some issues that need to be addressed before we can move forward. Please review the list below.

## Required changes :wrench: 
### [Mandatory] :red_circle: 

- [x] Style doesn't match original desing
- [x] The widht of the middle column
- [x] Debate section looks disproportionated
- [x] My turn section is duplicated, and elements don't have the correct size
image
- [x] the free sign up element doesn't match the style and layout.
**Clone**
image
**Original**
image
- [x] Sections missing or missing elements like titles are Culture and Travel, More Stories and Latest News, 
- [x] Layout, style, dimensions don't match also in the Tablet view.
image
image
image
- [ ] The order in Mobile view of the elements in the column doesn't match the roginal
- [ ] Also the style, size and don't match the original
image
image


## Keep up the great work :muscle: and Happy Coding :smiley: :computer:!!


<div>
  <div style="width:50px; height: 50px" align="center">
    <a href="https://avatars.githubusercontent.com/u/25774518?s=400&u=ae20ddf2d6fb3cae9e2f4044afe3b93d19e5b69e&v=4" alt="Julio Añoveros" title="@jubaan" width="100px" heigh="100px">
    </a>
  </div>
  </br>
  <p align="center">
    Julio Añoveros | @jubaan</br></br>
    <b>Full-stack Web Developer</b> and</br>
    Code Reviewer at <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a>:rocket:
  </p>
  <p align="center">
    <b>Feel stuck?</b></br>
    First breath... just do it.</br>
    Take 15-20 min away from the computer,</br>
    stretch, meditate, do some exercise, drink water,</br>
    or do whatever is necessary to distract your mind</br> 
    from the obstacle.</br></br> 
    Sometimes you just need to get away</br>
    to see things with a new perspetive.
  </p>
  <p align="center">
    <b>Need help?</b></br>
    Send me a DM in <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> Students Slack,</br>
    you'll find me as @jubaan.
  </p>
  </br>
  <p align="center">
    <q>A <b>great programer</b> is not the one that knows it all,</br>
    but the one that ask a lot of question.</q></br></br>
    <b>Don't hesitate to ask for help!</b>
  </p>
</div>

File: style.css [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+}
+
+.bold-link {
+  font-weight: 700;
+  font-size: 1.5em;
+}
+
+.editions {
+  flex-wrap: wrap;
+}
```
Comment (voscarmv) [inline style.css]: - As mentioned earlier:
> Use bootstrap classes for columns, flex, flex-direction, justify-content, align-content, display, position, font-weight, text-align... etc. Review the style.css file and remove all CSS rules that have a Bootstrap class and is useful for the project.

File: style.css [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    display: none;
+  }
+}
+
+@media (max-width: 450px) {
+  .footer-column {
+    display: flex;
+    flex-direction: column;
+  }
+}
```
Comment (voscarmv) [inline style.css]: - As mentioned earlier:
> Use bootstrap classes for columns, flex, flex-direction, justify-content, align-content, display, position, font-weight, text-align... etc. Review the style.css file and remove all CSS rules that have a Bootstrap class and is useful for the project.

**Review by voscarmv — CHANGES_REQUESTED** [review]
Hi. I accidentally deleted this review and was not able to undo the deletion, but here's a set of screenshots that I was able to recover from the edit history of this github comment. You should be able to use these as the review:

image
image
image
image
image

### Submit these changes and request a new code review! 😄 👍 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Comment by karvel-code** [comment]
Hello, We have made the necessary changes required by our reviewers. However, we have done our best to implement a bootstrap class for flex in the footer section but it is not leading to responsive results. Please advise.
Bishoy and Elvis.

**Review by jubaan — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

There are some issues that need to be addressed before we can move forward. Please review the list below.

## Required changes :wrench: 
### [Mandatory] :red_circle: 

- [x] Subscribe and Newsletter section should have the same height.
- [x] The Magazine Images in the Subscribe section are distorted, this should be fixed by working on the previous issue. But if it doesn't try `background-size: cover`
- [x] Use Bootstrap classes instead of custom CSS rules, please review the inline comments previously commented by other reviewers. 
  - [x] Display
  - [x] Positions
  - [x] Flex direction
  - [x] Flex wrap
  - [x] Bold font
  - [x] Text alignment
  - [x] Text decoration
  - [x] background color (white, black)
- [x] Remove the `!important` from the CSS styelsheet
- [x] Navbar menu elements should take the whole available space
image


## Keep up the great work :muscle: and Happy Coding :smiley: :computer:!!



<div>
  <div style="width:50px; height: 50px" align="center">
    <a href="https://avatars.githubusercontent.com/u/25774518?s=400&u=ae20ddf2d6fb3cae9e2f4044afe3b93d19e5b69e&v=4" alt="Julio Añoveros" title="@jubaan" width="100px" heigh="100px">
    </a>
  </div>
  </br>
  <p align="center">
    Julio Añoveros | @jubaan</br></br>
    <b>Full-stack Web Developer</b> and</br>
    Code Reviewer at <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a>:rocket:
  </p>
  <p align="center">
    <b>Feel stuck?</b></br>
    First breath... just do it.</br>
    Take 15-20 min away from the computer,</br>
    stretch, meditate, do some exercise, drink water,</br>
    or do whatever is necessary to distract your mind</br> 
    from the obstacle.</br></br> 
    Sometimes you just need to get away</br>
    to see things with a new perspetive.
  </p>
  <p align="center">
    <b>Need help?</b></br>
    Send me a DM in <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> Students Slack,</br>
    you'll find me as @jubaan.
  </p>
  </br>
  <p align="center">
    <q>A <b>great programer</b> is not the one that knows it all,</br>
    but the one that ask a lot of question.</q></br></br>
    <b>Don't hesitate to ask for help!</b>
  </p>
</div>

**Comment by jubaan** [comment]
> Hello, We have made the necessary changes required by our reviewers. However, we have done our best to implement a bootstrap class for flex in the footer section but it is not leading to responsive results. Please advise.
> Bishoy and Elvis.

Please is better if you reach out through Slack you can find me as @jubaan.

**Review by SpaYco — APPROVED** [review]
# Project Approved ✔️ 

thank you for making the required changes, your project is now approved!

### please don't forget to merge to master, good luck with the next projects!


Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Pull request merged**